pkg/ansible/runner.go

```diff
@@ -0,0 +1,201 @@
+	_, err := r.createInventoryConfigMap(inventory, vars)
+	if err != nil {
+		return err
+	}
+
+	jobName := "openshift-ansible-test-job"
+	env := []kapi.EnvVar{
+		{
+			Name:  "INVENTORY_FILE",
+			Value: "/ansible/inventory/hosts",
+		},
+		{
+			Name:  "PLAYBOOK_FILE",
+			Value: playbook,
+		},
+		{
+			Name:  "ANSIBLE_HOST_KEY_CHECKING",
+			Value: "False",
+		},
+		{
+			Name:  "OPTS",
+			Value: "-vvv --private-key=/ansible/ssh/privatekey.pem -e @/ansible/inventory/vars",
+		},
+		{
+			Name: "AWS_ACCESS_KEY_ID",
+			ValueFrom: &kapi.EnvVarSource{
+				SecretKeyRef: &kapi.SecretKeySelector{
+					LocalObjectReference: kapi.LocalObjectReference{Name: awsCredentialsSecret},
+					Key:                  "aws_access_key_id",
+				},
+			},
+		},
+		{
+			Name: "AWS_SECRET_ACCESS_KEY",
+			ValueFrom: &kapi.EnvVarSource{
+				SecretKeyRef: &kapi.SecretKeySelector{
+					LocalObjectReference: kapi.LocalObjectReference{Name: awsCredentialsSecret},
+					Key:                  "aws_secret_access_key",
+				},
+			},
+		},
+	}
+	runAsUser := int64(0)
+	sshKeyFileMode := int32(0600)
+	playbookPath := path.Join(openshiftAnsibleContainerDir, playbook)
+	podSpec := kapi.PodSpec{
+		DNSPolicy:          kapi.DNSClusterFirst,
+		RestartPolicy:      kapi.RestartPolicyNever,
+		ServiceAccountName: openshiftAnsibleServiceAccount,
+		//HostNetwork:        true,
+
+		Containers: []kapi.Container{
+			{
+				Name:  jobName,
+				Image: r.Image,
+				Env:   env,
+				SecurityContext: &kapi.SecurityContext{
+					RunAsUser: &runAsUser,
+				},
+				VolumeMounts: []kapi.VolumeMount{
+					{
+						Name:      "inventory",
+						MountPath: "/ansible/inventory/",
+					},
+					{
+						Name:      "sshkey",
+						MountPath: "/ansible/ssh/",
+					},
+				},
+				//Command: []string{"sleep", "1000000"},
+
+				// TODO: drop this once https://github.com/openshift/openshift-ansible/pull/6320 merges, the default run script should then work, minus the vars file :(
+				Command: []string{"ansible-playbook", "-i", "/ansible/inventory/hosts", "--private-key", "/ansible/ssh/privatekey.pem", playbookPath, "-e", "@/ansible/inventory/vars"},
+			},
+		},
+		Volumes: []kapi.Volume{
+			{
+				// Mounts both our inventory and vars file.
+				Name: "inventory",
+				VolumeSource: kapi.VolumeSource{
+					ConfigMap: &kapi.ConfigMapVolumeSource{
+						LocalObjectReference: kapi.LocalObjectReference{
+							Name: inventoryConfigMap,
+						},
+					},
+				},
+			},
+			{
+				Name: "sshkey",
+				VolumeSource: kapi.VolumeSource{
+					Secret: &kapi.SecretVolumeSource{
+						SecretName: sshPrivateKeySecret,
+						Items: []kapi.KeyToPath{
+							{
+								Key:  "ssh-privatekey",
+								Path: "privatekey.pem",
+								Mode: &sshKeyFileMode,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	completions := int32(1)
+	deadline := int64(60 * 60) // one hour for now
+
+	meta := metav1.ObjectMeta{
+		GenerateName: jobNamePrefix,
+		Namespace:    operatorNamespace,
```

Comment:
It may be useful to add labels to correlate the job to a cluster resource.

Reply:
Definitely. We will some one to associate the jobs with the cluster for no other reason than ensuring that we are cleaning everything up when the cluster is deleted. @csrwng, is there a preference on using labels versus using owner references? For ClusterController, I set the Cluster as the controlling owner of each NodeGroup created for the Cluster.

Reply:
@staebler it will depend on whether the jobs/configmaps will live in the same namespace as the cluster resource or not. An owner ref is definitely preferable, but I don't think you can have owner refs across namespaces.

Reply:
I'm still hoping that we can have all the resources needed to stand up a cluster created in the same namespace as the cluster.

Reply:
See my comment about the service account. It'd be great if we don't need a runAsAny SCC